src/components/labeledInput.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { ReactElement } from 'react'
+import { TextField, Box, Checkbox, FormControlLabel } from '@material-ui/core'
+import { Control, Controller, RegisterOptions } from 'react-hook-form'
+import InputMask from 'react-input-mask'
+import locationPostalCodes from '../data/locationPostalCodes'
+
+interface Errors {
+  [key: string]: {type: string}
+}
+
+type Register = (rules?: RegisterOptions) => any
+
+interface PatternConfig {
+  mask?: string
+  regexp?: RegExp
+  description?: string
+}
+
+interface LabeledInputProps {
+  strongLabel?: string
+  patternConfig?: PatternConfig
+  label: string
+  register: Register
+  required: boolean
+  name: string
+  errors: Errors
+  defaultValue?: string
+}
+
+const getError = (errors: Errors, name: string): string | undefined => {
+  let result
+  if (Object.prototype.hasOwnProperty.call(errors, name)) {
+    const r = errors[name]
+    result = r.type
+  }
+  return result
+}
+
+const isError = (errors: Errors, name: string): boolean => getError(errors, name) !== undefined
+
+export function LabeledInput (props: LabeledInputProps): ReactElement {
+  const { strongLabel, label, register, required, patternConfig = {}, name, errors, defaultValue } = props
```

Comment:
can be replaced with
`export function LabeledInput({ strongLabel = '', patternConfig, label, register, required, name,errors, defaultValue}: LabeledInputProps): ReactElement {`

Reply:
I kind of want to keep it this way? A combination of readability of the type signature and keeping the line from getting too long seems better to me than being concise with destructuring.